### PR TITLE
RAP-4127 and RAP-4116

### PR DESCRIPTION
**Pull request by scarabus**
RAP-4116 Integration tests for search API.
Includes a helper class to do simple login, repo creation and deletion etc.
Also includes a Query with Retry option for searches that may take varying times.

### CODE REVIEW

**Comment by jonathan-major** [comment]
confused by the pr. is it for integration tests or a code fix?

File: Apps/RaptureIntTests/src/test/java/rapture/api/checkout/SearchApiIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+     *
+     * @param RaptureURL
+     *            Passed in from <env>_testng.xml suite file
+     * @param RaptureUser
+     *            Passed in from <env>_testng.xml suite file
+     * @param RapturePassword
+     *            Passed in from <env>_testng.xml suite file
+     * @return none
+     */
+    @BeforeClass(groups = { "mongo" })
```
Comment (jonathan-major) [inline Apps/RaptureIntTests/src/test/java/rapture/api/checkout/SearchApiIntegrationTest.java]: the execution groups need to be co-ordinated with Jeremy.  Cant have different groups across the test classes.

What is the thought on running the tests against the various Rapture configurations e.g. FTS off or on

File: Apps/RaptureIntTests/src/test/java/rapture/api/checkout/IntegrationTestHelper.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
+package rapture.api.checkout;
```
Comment (jonathan-major) [inline Apps/RaptureIntTests/src/test/java/rapture/api/checkout/IntegrationTestHelper.java]: no MIT license

File: Libs/RaptureAPI/src/main/java/rapture/common/BlobUpdateObject.java [hunk 3]
```diff
@@ -36,5 +36,6 @@ public BlobContainer getPayload() {
     @Override
     public void setPayload(BlobContainer payload) {
         this.payload = payload;
+        setMimeType(payload.getHeaders().get(ContentEnvelope.CONTENT_TYPE_HEADER));
```
Comment (jonathan-major) [inline Libs/RaptureAPI/src/main/java/rapture/common/BlobUpdateObject.java]: Is this related to RAP-4127

File: Libs/RaptureCore/src/main/java/rapture/kernel/SeriesApiImpl.java [hunk 4]
```diff
@@ -158,6 +158,20 @@ public void deletePointsFromSeriesByPointKey(CallingContext context, String seri
         RaptureURI internalURI = new RaptureURI(seriesURI, Scheme.SERIES);
         SeriesRepo repo = getRepoOrFail(internalURI);
         repo.deletePointsFromSeriesByColumn(internalURI.getDocPath(), columns);
+        if (SearchPublisher.shouldPublish(getConfigFromCache(internalURI.getAuthority()), internalURI)) {
+            // Need to delete points from ES.
+            // Drop the old series and re-add all the points that remain.
+            // Is there a better way?
+            SearchPublisher.publishDeleteMessage(context, getConfigFromCache(internalURI.getAuthority()), internalURI);
```
Comment (jonathan-major) [inline Libs/RaptureCore/src/main/java/rapture/kernel/SeriesApiImpl.java]: what is the performance overhead of doing this?

File: Libs/RaptureCore/src/main/java/rapture/kernel/SeriesApiImpl.java [hunk 5]
```diff
@@ -158,6 +158,20 @@ public void deletePointsFromSeriesByPointKey(CallingContext context, String seri
         RaptureURI internalURI = new RaptureURI(seriesURI, Scheme.SERIES);
         SeriesRepo repo = getRepoOrFail(internalURI);
         repo.deletePointsFromSeriesByColumn(internalURI.getDocPath(), columns);
+        if (SearchPublisher.shouldPublish(getConfigFromCache(internalURI.getAuthority()), internalURI)) {
+            // Need to delete points from ES.
+            // Drop the old series and re-add all the points that remain.
+            // Is there a better way?
```
Comment (jonathan-major) [inline Libs/RaptureCore/src/main/java/rapture/kernel/SeriesApiImpl.java]: Is the comment answered? If so, then delete it.

File: Apps/RaptureIntTests/src/test/java/rapture/api/checkout/SearchApiIntegrationTest.java (hunk 1 above)
Comment (scarabus) [inline Apps/RaptureIntTests/src/test/java/rapture/api/checkout/SearchApiIntegrationTest.java]: These are search based tests; they will fail with FST turned off.
Will work with Jeremy on defining the set of group names.

**Comment by scarabus** [comment]
The PR is for RAP-4116 and RAP-4127, which was discovered by the test.
The pull request just took the first line of the commit as subject. I've fixed that.

File: Libs/RaptureCore/src/main/java/rapture/kernel/SeriesApiImpl.java (hunk 4 above)
Comment (scarabus) [inline Libs/RaptureCore/src/main/java/rapture/kernel/SeriesApiImpl.java]: Difficult to say. It depends on the number of points in the series - how many are deleted and how many remain. If you're deleting a small number of points from a large series it's probably highly inefficient. If you're using the method to wipe out the series it's more efficient than deleting the points. We don't have a method to determine the number of points in a series without querying the entire series, which makes it difficult to judge. Also we don't have any idea how heavily used this method is likely to be.

**Pull request closed without merging**